### PR TITLE
Make indexing for authoring always async

### PR DESCRIPTION
**Pull request by jdrossl**
https://github.com/craftercms/craftercms/issues/3152

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v1/deployment/PreviewDeployerImpl.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ public void subscribeToPreviewSyncEvents() {
 
     @EventListener(EVENT_PREVIEW_SYNC)
     public void onPreviewSync(PreviewEventContext context) {
-        doDeployment(context.getSite(), ENV_AUTHORING, context.isWaitTillDeploymentIsDone());
```
Comment (avasquez614) [inline src/main/java/org/craftercms/studio/impl/v1/deployment/PreviewDeployerImpl.java:88]: Shouldn't we remove this property from the context too?

**Review by avasquez614 — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/impl/v1/deployment/PreviewDeployerImpl.java (hunk 1 above)
Comment (jdrossl) [reply src/main/java/org/craftercms/studio/impl/v1/deployment/PreviewDeployerImpl.java:88]: Sometimes it is set to `false`, not sure what the explanation is for each use case: db sync & cluster sync

Comment (avasquez614) [reply src/main/java/org/craftercms/studio/impl/v1/deployment/PreviewDeployerImpl.java:88]: Ok, I thought the flag wasn't used somewhere else, my bad.

Comment (sumerjabri) [reply src/main/java/org/craftercms/studio/impl/v1/deployment/PreviewDeployerImpl.java:88]: I say we keep the option to wait.

**Review by avasquez614 — APPROVED** [review]

**Pull request merged**